content/200-concepts/100-components/02-prisma-client/053-middleware/100-soft-delete-middleware.mdx

```diff
@@ -598,3 +604,15 @@ prisma.post.findMany({ where: { includeDeleted: true } })
 > **Note**: You would still need to write middleware.
 
 We **✘ do not** recommend this approach as it pollutes the schema with fields that do not represent real data.
+
+## Limitations
+
+The soft delete sample on this page is not a full implementation. In the options above, we explain some of the limitations. Other limitations include the following:
+
+- `findMany` with options such as `some` and `none` does not work
+- Most nested queries, such as `findMany({ where: {}, include: {}})` and `update({data: { posts: { deleteMany: { } } } })` do not work
+- [Cascading deletes](/guides/database/advanced-database-tasks/cascading-deletes) do not work
+- You cannot update related tables with `updateMany`
+- You cannot enforce [referential actions](/concepts/components/prisma-schema/relations/referential-actions)
+- You cannot filter deleted relations
+- You cannot use the sample in conjunction with a conditional unique index
```

Comment:
Are there any issues for these limitations?

Reply:
As far as I'm aware, we have one engineering issue to improve soft delete, which is here: https://github.com/prisma/prisma/issues/10559, which I'm monitoring. The discussion in that ticket contains most of the items in the list above, and I've pulled in others from one or two other tickets I found.

That said, the issues in the list were reported by users, in particular the discussion in the issue above. Some of these are reported by more than one user, but really, we need to confirm them. I'm starting to regret adding this list now, because we don't know for sure that all of these are issues, so next week I will remove them and just leave the warnings at the top of the page.